src/web/components/SlotBodyEnd.tsx

```diff
@@ -168,96 +163,60 @@ const MemoisedInner = ({
             contributionsServiceUrl,
             isSensitive,
         });
-        getBodyEnd(contributionsPayload, `${contributionsServiceUrl}/epic`)
+
+        window.automat = {
+            react: React,
+            emotionCore,
+            emotionTheming,
+            emotion,
+        };
+
+        getBodyEnd(
+            contributionsPayload,
+            `${contributionsServiceUrl}/epic?dataOnly=true`,
+        )
             .then(checkForErrors)
             .then(response => response.json())
             .then(json => {
-                if (json.data) {
-                    setData({
-                        slot: {
-                            html: json.data.html,
-                            css: json.data.css,
-                            js: json.data.js,
-                            meta: json.data.meta,
-                        },
-                    });
-
-                    sendOphanEpicEvent('INSERT', json.data.meta);
+                if (!json.data) {
+                    return;
                 }
-            })
-            .catch(error =>
-                window.guardian.modules.sentry.reportError(
-                    error,
-                    'slot-body-end',
-                ),
-            );
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, []); // only ever call once (we'd rather fail then call the API multiple times)
 
-    useEffect(() => {
-        // This won't be true until we've successfully fetched the data and
-        // rendered the epic (because of how we're wiring up the ref below). And
-        // because of the way the hook behaves, it'll only ever go from false ->
-        // true once.
-        if (hasBeenSeen) {
-            const meta = data?.slot?.meta;
+                const { meta, module } = json.data;
+                const epic = import(/* webpackIgnore: true */ module.url);
 
-            if (meta) {
-                // Add a new entry to the view log when we know an Epic is viewed
-                logView(meta.abTestName);
-                sendOphanEpicEvent('VIEW', meta);
-            }
-        }
-    // The 'data' object used in the hook never changes after 'hasBeenSeen'
-    // is set to true, so we're intentionally leaving it out of the deps array.
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [hasBeenSeen]);
+                // TODO, we need to reconsider event handling, as this is all
+                // highly coupled at the moment (better would be data
+                // attributes).
+                const props = {
```

Comment:
I do not see the point in splitting this out, nor in having the variable `epic` defined , suggestion:

```
import(/* webpackIgnore: true */ module.url).then(() => {
    setEpicMeta(meta);
    setEpicProps({
        ...module.props,
        onReminderOpen: sendOphanReminderOpenEvent,
    });
    setEpic(() => epicModule.ContributionsEpic); // useState requires functions to be wrapped
    sendOphanEpicEvent('INSERT', meta);
    // eslint-disable-next-line no-console
}).catch(error => console.log(`epic - error is: ${error}`));

```

Reply:
it is also worth adding some performance analysis here, refer to https://github.com/guardian/dotcom-rendering/blob/master/src/web/components/CMP.tsx#L20 for how to implement it in imports

Reply:
Ah great, both really useful points. I was wondering about how to track performance.

Reply:
@liywjl does the perf tracking get sent somewhere?

Reply:
it gets picked up by [speed curve](https://speedcurve.com/) its hidden deep in one of the reports, best to ask @gtrufitt for access

Reply:
Yeah, if you add user timing metrics, they'll be captured on runs and we can add a graph to speedcurve. I am going to create a new 'user timings' dashboard so we can get an overview of our module perf

Reply:
awesome sounds good. Thanks both.